teamcity/datadog_checks/teamcity/check.py

```diff
@@ -0,0 +1,43 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from copy import deepcopy
+from urllib.parse import urlparse
+
+from datadog_checks.base import OpenMetricsBaseCheckV2, is_affirmative
+
+from .metrics import METRIC_MAP
+
+
+class TeamCityCheckV2(OpenMetricsBaseCheckV2):
```

Comment:
If I understand correctly the intent here is that the user would define two instances, one openmetrics and one not openmetrics to collect events and service checks. In that sense I don't think naming this V2 is the best idea because it is not meant to replace the other implementation but to complement it. I think we should name this one TeamCityOpenmetrics and rename the other one as TeamCityRest (for example). In the same vein if this is not intendended to ever become the main entry point for the check I don't think calling this `check` conveys the right idea, I think we can rename the files to to teamcity_openmetrics and teamcity_rest

Reply:
That makes sense, I've renamed the files and changed `check.py` to be the entry point